src/freehand_selection.jl

```diff
@@ -1,61 +1,188 @@
 function init_freehand_select(ctx::ImageContext)
     c = ctx.canvas
     enabled = Signal(true)
-    dragging = Signal(false)
+    drawing = Signal(false)
+    moving = Signal(false)
+    diff = Signal(XY(NaN,NaN))
 
     dummybtn = MouseButton{UserUnit}()
     
     sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
         if !ispolygon(value(ctx.points)) || !isinside(Point(btn.position), Point.(value(ctx.points))) # prevents conflict with init_move_polygon
```

Comment:
You could simplify a fair bit of what's below by either (1) writing methods of `ispolygon` and other functions that accept a `Signal` input (and call `value` for you), or (2) before this `if` block just write `points = value(ctx.points)` and then use `points` everywhere below.